tests/ragged_tests.py

```diff
@@ -699,6 +699,15 @@ def test_subset_by_rows(self):
         self.assertTrue(all(ds_sub["id"] == [1, 2]))
         self.assertTrue(all(ds_sub["rowsize"] == [5, 4]))
 
+    def test_subset_callable(self):
+        func = (
+            lambda arr: ((arr - arr[0]) % 2) == 0
+        )  # test keeping obs every two time intervals
+        ds_sub = subset(self.ds, {"time": func})
```

Comment:
I'm not sure how the test is passing, you are not passing a `dim_name` as I see `ds` has dimensions of `obs` and `traj`... Am I missing something?

Would it be possible to set the default as one of those two instead of xarray default `dim_0` when dimension names are not set.

Reply:
`dim_name` is passed to `_mask_var` within `subset`, so we don't need to specify it here.

For your second point, I had more or less the same question. I made `row_size` and `dim_name` arguments as optional because they are not always useful. But at the same time, `_mask_var` is for now always used with the four arguments (except in the docstring examples). Maybe default values could be removed.

Reply:
Ok I see now. Since the docstring are written like let's keep them optional. And it is not supposed to be called by the users directly anyway.